### DISCUSSION TITLE
Welcome to Beehive Discussions!

### DISCUSSION THREAD

**Discussion by pradeeban** [KathiraveluLab/Beehive#1] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by mdxabu** [comment]
Hello @pradeeban ,

I hope you are doing well. I am Mohamed Abdullah and I am writing to let you know about my interest in your [GSoC 2024 Project[5]](https://github.com/uaanchorage/GSoC). I find it particularly intriguing and would love to contribute to this project. Could you please provide me with more details about it? Could you please provide me information on the environment setup for Java and Python, as well as the build tools like version and so on.?

I am looking forward to hearing from you.

Best regards,
F Mohamed Abdullah

**Comment by Vamsi995** [comment]
Hello @pradeeban,

I am Alisetti Sai Vamsi, a Computer Science graduate from IIT Palakkad. With a profound interest in leveraging technology for societal impact, particularly in critical areas like healthcare, the Beehive project resonates deeply with both my skill set and personal values. 

My strong background in Python/Java programming and expertise in database management, coupled with a passion for utilizing technology to address societal challenges, positions me well to contribute meaningfully to Beehive. Furthermore, the project's focus on supporting marginalized communities, such as the unhoused, through digital advocacy and art representation deeply resonates with my belief in the transformative power of technology to uplift and empower vulnerable populations.

Can you please provide me with the framework/structure of the project and possibly the starting steps to take?

Regards,
Alisetti Sai Vamsi

**Comment by VithuThevan** [comment]
Hi @pradeeban,

I am Vithursan Thevendran, a final year IT undergraduate from the University of Moratuwa, reading B.Sc Hons. in Information Technology. I found GSoC project 5 to be an extreme turning point in the field of Healthcare. With my knowledge and expertise in Java and Python, I am ready to contribute to this project with you all.

Looking forward to work with you all!

Thank you.

Regards,
Vithursan Thevedran

**Comment by MohamedNasser8** [comment]
Hello @pradeeban,

I am Mohamed Nasser, a Biomedical Engineering student at Cairo University. Throughout my academic journey, I've engaged extensively with various forms of medical data, including images and multi-channel time series. This exposure has equipped me with a deep understanding of the challenges surrounding data storage and retrieval in the medical domain.
Proficient in Python, Java, and several other programming languages, I am excited about the opportunity to contribute to the Beehive project.
I am eager to collaborate with the University of Alaska Anchorage Departments of Computer Science and Human Services.
So can you tell me from where to start?

**Comment by UncleWeeds** [comment]
hey @pradeeban,

I am Abhiram a sophomore at NITT, My primary domain is devOps, other than that I mainly work in backend, cybersecurity, and Blockchain. I really like the philosophy behind open source and decentralization but due to prior commitments I was not able to work on them during my first year but now I have been active in the open source community for the past 7 to 8 months, other than that right now I am actively working with my club to improve my collage tech infrastructure and building a project for a startup that's being incubated by NTU Singapore. 

So I chose this project specifically because I have primarily used MongoDB for many of my club's work and personally have made PRs for open source orgs for mongo also I'm passionate about using technology to make a real-world impact, particularly in healthcare and human services, and I'm excited to explore how digital platforms like Beehive can enhance healthcare delivery for vulnerable communities. I look forward to learning from and contributing to this community, sharing ideas, and engaging in discussions that help us grow together. 
would love to get ur advice regarding this project

**Comment by mayankmehta8** [comment]
Hello @pradeeban 

I am Mayank, a third year student pursuing the field of AI and DS. I have a keen interest in working with data and am familiar with a lot of data preprocessing and extracting techniques using python. I also have work experience working with SQL databases during my previous internship period. 
I have a keen interest in the field of healthcare (mental health) and creating societal impact as I have also volunteered in AIESEC before. 
The project beehive has greatly intrigued me and I can see a lot of value in its completion and would love to be a part of it by contributing.
Lastly, my goal of joining this project is to learn from mentors and I hope that beehive will be a great success. 
Please give me some advice regarding how I can contribute to this project and I would love to know more about it as well.

Regards,
Mayank Mehta

**Comment by ChathuraAbeygunawardhana** [comment]
@pradeeban ,

I am Chathura Abeygunawardhana, a second year IT undergraduate from the University of Moratuwa. I got a good grasp on Java / Python and web development as well. I found The Beehive project fascinating, and I look forward to adding my contribution on this project. Looking forward to work with you all!

Thank you.

Regards,
Chathura Abeygunawardhana

**Comment by yanhuojunjun** [comment]
Hi @pradeeban ,
I am Mingjun Li, a junior student majoring in Computer Science and Technology at Sun Yat-sen University. I am reaching out to express my sincere interest in the Beehive project listed for GSoC.  
After reviewing the project details, I am particularly enthusiastic about the opportunity to contribute to the the field of healthcare. I am proficient in Python, and have experience of constructing web by Django, which makes me well-suited for this task. Furthermore, I am willing to dedicate my time and effort to this project, ensuring that it receives the attention and dedication it deserves.

**Comment by JiuYou2020** [comment]
Hi @pradeeban,

I'm Kun Gao, a junior student majoring in Computer Science and Technology at Northeastern University. I'm particularly excited about the GSoC 2024 project [5] and after reviewing its requirements, I believe I have the skills necessary to contribute to Beehive's development. I'm proficient in Java, MySQL, and MongoDB, and I've developed web services under the guidance of my school's faculty on multiple occasions.

I am eager to make meaningful contributions to the Beehive project and ensure its robustness. Thank you for considering my candidacy.

Best regards,
Kun Gao

**Comment by 01bps** [comment]
Hi @pradeeban and everyone! 

I am Bagwan, currently in my third year of Electrical Engineering at RGIPT. I am new to open-source contributions, with experience in Python and Java, as well as the MySQL database. I have some exposure to image training, classification, and recognition, which I explored in a project: [.](https://github.com/01bps/Face_Recognition_DIC_P)  https://github.com/01bps/Face_Recognition_DIC_P

From the list of offered project ideas, I would like to work on Project Idea # 3, Beehive. 
I am looking forward to learn from this community.

Thank you.

Best regards,
Bagwan

**Reply by mdxabu** [reply]
Please check the repository where we started[0] and nearly finished the project prototype. You can review it and work on the documentation.
[0]: https://github.com/KathiraveluLab/Beehive

**Reply by 01bps** [reply]
Sure, 
I'll start by thoroughly reviewing the project and then work on improving the documentation.

**Reply by 01bps** [reply]
I have successfully set up the project locally and have been reviewing the documentation in detail. I tested the features in the Implementation Details, and everything I tried is functioning well. While reviewing, I noticed two broken links in the documentation (fixed in my forked repository). Additionally, if needed, the documentation can be made more beginner-friendly, although it worked well for me. Could you please advise on the next steps or tasks I should prioritise?

Looking forward to your suggestion!

Best regards,
Bagwan

**Reply by mdxabu** [reply]
If you fix those broken links, well-documented markdowns are always welcome. Feel free to open pull requests.

Regards,
Mohamed Abdullah F

**Reply by 01bps** [reply]
I hope you're doing well. This is a gentle reminder about the PR I submitted for the documentation updates. Please let me know if there’s anything I should address or improve in it.

Thank you!

**Reply by 01bps** [reply]
I’ve added an update to allow multiple images with shared metadata to be uploaded at once, making it more user-friendly for researchers and marginalised folks. However, I haven’t specified the acceptable image formats. Should we define supported file types explicitly, and if so, what formats should we allow?

Looking forward to your suggestion!

Best regards,
Bagwan

**Comment by ishaanxgupta** [comment]
Hi @pradeeban @mdxabu 
My name is Ishaan Gupta, and I’m currently pursuing a B.Tech in Computer Science and Data Science from NSUT, New Delhi, India. I am familiar with technologies like Python (Flask, FastAPI, Django), SQL & MongoDB, MERN stack, Redis, and WebSockets. I have also interned at startups, working on FastAPI and AI solutions.
Here is my Github:- https://github.com/ishaanxgupta

I’m very interested in contributing to Project #3 Beehive as part of Google Summer of Code (GSoC) 2025 and would like to propose a few ideas for potential improvements to the project:

1. The UI can be enhanced by integrating Tailwind CSS or Bootstrap, creating a responsive and aesthetically pleasing experience that aligns with the project’s mission.
2. Instead of storing images in the static folder, I suggest saving them as Base64 strings or binary data in the database (e.g., using MongoDB’s GridFS).
3. Allow contributors to add tags (e.g., “unhoused,” “recovering alcoholic”) when submitting their art. This can help analyze and report on contributions and offer better support for specific groups.
4. Enable both users and admins to download images in specific formats to improve the ease of content retrieval and sharing.
5. Enhance the database to allow for user interactions like likes, comments, and shares, promoting engagement and interaction within the community.

I would be excited to contribute and refine these ideas based on feedback. Please let me know your thoughts or any other features you’d like me to focus on.

Thank you for your time and consideration!

Best regards,
Ishaan Gupta

**Comment by ishaanxgupta** [comment]
@pradeeban @mdxabu 
Waiting for your suggestions

Than

**Comment by ishaanxgupta** [comment]
Hi @pradeeban @mdxabu,
I have tried to implement some features which were mentioned in the issue as well as README.md file. I have raised a PR #27 

Here is the code for the same. 
https://github.com/ishaanxgupta/Beehive-alaska

Please give your feedback.

**Comment by chandrikarj** [comment]
Hi @pradeeban @mdxabu,

I am Chandrika Singh Jadon, and I’m currently pursuing a B.Tech in Computer Science from BITS Pilani, Goa Campus, India. I have experience with technologies such as Python (Flask, FastAPI, Django), SQL & MongoDB, OpenCV, and Machine Learning frameworks. I have previously worked on projects involving biomedical signal extraction using computer vision, modeling state machines for digital systems, and communication protocols for autonomous vehicles.

Here is my GitHub: https://github.com/chandrikarj

I’m excited to contribute to Project #3 Beehive and have recently submitted a PR : https://github.com/KathiraveluLab/Beehive/pull/29 that introduces image search functionality and enhances metadata display on the profile page. I look forward to your valuable feedback and any suggestions for further improvements.

Thank you for your time and consideration.

Best regards,
Chandrika Singh Jadon

**Comment by IVedantD** [comment]
Discrepancy in UI Design Between Local Setup and GitHub Issue #150 

Hello Beehive Development Team,

I've set up the Beehive project locally, but my UI differs from what's shown in Issue #150. Could you guide me on: 

Branch Information:Which branch contains the latest UI changes? 
Update Instructions: How can I align my local environment with the current UI design?  
 
 I've attached a comparison image that shows some of the differences. 
 
 local UI : 
![image](https://github.com/user-attachments/assets/994932db-ef64-4bf1-8448-7aa56be85237)
![image](https://github.com/user-attachments/assets/41b8e8b0-4a7c-4f01-9d16-78a848a35634)

Reference UI:
![image](https://github.com/user-attachments/assets/a7cd5068-bbf3-4f9d-a8e2-2d77370b2614)

**Comment by ishaanxgupta** [comment]
Hi @IVedantD 
Please refer to the dev branch for latest changes
ig you are refering to the main branch

**Comment by IVedantD** [comment]
@ishaanxgupta 
Thanks for your help! I switched to the dev branch and pulled the latest changes. The UI now looks correct.

**Reply by pradeeban** [reply]
awesome

**Comment by IffatHafsa** [comment]
Hi Iffat Hafsa , Do we need both java and python ?

**Reply by pradeeban** [reply]
No. Python should be sufficient.

**Comment by ishaanxgupta** [comment]
Hi @pradeeban,
I emailed you and David a draft proposal. Please have a look whenever you get time.
Will be waiting for your suggestions and feedback.
Do let me know if the same was not received at your end, will check for the correct emails, Although I hope I have entered them correctly
Thanks

**Reply by pradeeban** [reply]
Received it. It is well-written. Thanks!

**Comment by MOSTAFA-MANSOUR72** [comment]
HI @pradeeban ,
I am Mostafa Mansour, And I am a comuter science student at my graduation year. I have experience in developing web applications using various technologies like spring boot ( Java ), flask ( python), mysql and more. I am passionate about contributing and working on such an open source project like beehive via Gsoc 2025 enhancing my skills and get familiar with how software projects are being developed in the real world.

Best regards,
Mostafa Mansour

**Comment by MOSTAFA-MANSOUR72** [comment]
Hi @pradeeban,
I am wondering how to get the tasks i am supposed to complete to have better chances in gsoc 2025, is it related to posting and solving issues related to the project?

**Reply by pradeeban** [reply]
Hi and welcome @MOSTAFA-MANSOUR72!
We have a general [Rubrics](https://github.com/uaanchorage/GSoC/blob/main/Rubrics.md). But GSoC tends to be competitive. So, oftentimes, we end up not accepting many great proposals. (Because GSoC only gives certain number of slots to each organizations. The winning strategy is to go through the repositories carefully and try your best. Good luck.
